src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1449,10 +1449,13 @@ void ICanSaveModel.Save(ModelSaveContext ctx)
             var propertyInfos =
                 userType
                 .GetProperties(BindingFlags.Public | BindingFlags.Instance)
-                .Where(x => x.CanRead && x.CanWrite && x.GetGetMethod() != null && x.GetSetMethod() != null && x.GetIndexParameters().Length == 0);
+                .Where(x => x.CanRead && x.GetGetMethod() != null && x.GetIndexParameters().Length == 0);
 
             var memberInfos = (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
 
+            if (memberInfos.Length == 0 && dataSample == null)
```

Comment:
>dataSample [](start = 43, length = 10)

Maybe this should throw even if `dataSample` is not null?
If it is not null, then `TextLoader` will create a schema with two columns `Label` and `Features`, but it will not be able to read `Label` and `Features` from objects of type `TInput` - wouldn't it throw somewhere downstream in that case?

Reply:
I did not check I will see what happens in that case.

Reply:
I feel that if the user wants `TextLoader` to automatically load a `Label` and a `Features` column, they should use one of the non-generic overloads of the `CreateTextLoader` API. If they choose to use the generic overload, we should assume that their intent is to load the fields/properties in the class, and if `memberInfos` ends up being empty, then it means they didn't define the class properly, thus we should throw whether or not they provide a `dataSample`. What do you think?

---
In reply to: [290417237](https://github.com/dotnet/machinelearning/pull/3788#discussion_r290417237) [](ancestors = 290417237)

Reply:
I agree I did that after we discussed.